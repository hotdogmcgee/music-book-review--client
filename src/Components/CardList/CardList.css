.select-boxes {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
}

.select-box {
    height: 100px;
    border: 1px solid black;
    width: 140px;
    margin: 10px;
}


/* a bit of safety for extra small screens */
@media screen and (max-width: 330px) {
    .select-box {
        width: 135px
    }
}

@media screen and (min-width: 750px) {
    .select-box {
        height: 157px;
        width: 220px
    }
}

@media screen and (min-width: 1000px) {
    .select-box {
        height: 250px;
        width: 350px
    }
}