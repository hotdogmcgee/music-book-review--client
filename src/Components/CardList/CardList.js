import React from 'react'
import InstrumentCard from '../InstrumentCard/InstrumentCard'
import { Section } from '../Utils/Utils'
import './CardList.css'

export default function CardList(props) {
    const cardList = props.instruments.map(item => {
        return <div className='select-box'><InstrumentCard title={item.title}/></div>
    })
    return (
        <Section className='select-boxes'>
            {cardList}
        </Section>
    )
}