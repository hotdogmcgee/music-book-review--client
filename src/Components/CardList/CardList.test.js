import React from 'react';
import CardList from './CardList'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('CardList component', () => {
    const instrumentList = [
        { title: "guitar" },
        { title: "piano" },
        { title: "clarinet" },
        { title: "violin" }
      ];
    it('renders a CardList by default', () => {
        const wrapper = shallow(<CardList />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the CardList given props', () => {
        const wrapper = shallow(<CardList instruments={instrumentList} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})
