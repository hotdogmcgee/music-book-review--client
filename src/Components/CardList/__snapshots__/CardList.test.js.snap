// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component renders a CardList by default 1`] = `
<Section
  className="select-boxes"
/>
`;

exports[`CardList component renders the CardList given props 1`] = `
<Section
  className="select-boxes"
>
  <div
    className="select-box"
    key="0"
  >
    <InstrumentCard
      title="guitar"
    />
  </div>
  <div
    className="select-box"
    key="1"
  >
    <InstrumentCard
      title="piano"
    />
  </div>
  <div
    className="select-box"
    key="2"
  >
    <InstrumentCard
      title="clarinet"
    />
  </div>
  <div
    className="select-box"
    key="3"
  >
    <InstrumentCard
      title="violin"
    />
  </div>
</Section>
`;
